pkg/synthetictests/kubelet.go

```diff
@@ -177,40 +173,43 @@ func testContainerFailures(events monitorapi.Intervals) []*junitapi.JUnitTestCas
 
 func testConfigOperatorProbeErrorReadinessProbe(events monitorapi.Intervals) []*junitapi.JUnitTestCase {
 	const testName = "[sig-node] openshift-config-operator should not get probe error on readiness probe due to timeout"
-	return makeProbeTest(testName, events, "openshift-config-operator", probeErrorReadinessMessageRegExpStr, probeErrorReadinessEventThreshold)
+	return makeProbeTest(testName, events, "openshift-config-operator", probeErrorReadinessMessageRegExpStr, duplicateEventThreshold)
 }
 
 func testConfigOperatorProbeErrorLivenessProbe(events monitorapi.Intervals) []*junitapi.JUnitTestCase {
 	const testName = "[sig-node] openshift-config-operator should not get probe error on liveness probe due to timeout"
-	return makeProbeTest(testName, events, "openshift-config-operator", probeErrorLivenessMessageRegExpStr, probeErrorLivenessEventThreshold)
+	return makeProbeTest(testName, events, "openshift-config-operator", probeErrorLivenessMessageRegExpStr, duplicateEventThreshold)
 }
 
 func testConfigOperatorReadinessProbe(events monitorapi.Intervals) []*junitapi.JUnitTestCase {
 	const testName = "[sig-node] openshift-config-operator readiness probe should not fail due to timeout"
-	return makeProbeTest(testName, events, "openshift-config-operator", readinessFailedMessageRegExpStr, readinessFailedTimeoutEventThreshold)
+	return makeProbeTest(testName, events, "openshift-config-operator", readinessFailedMessageRegExpStr, duplicateEventThreshold)
 }
 
 func makeProbeTest(testName string, events monitorapi.Intervals, operatorName string, regExStr string, eventFlakeThreshold int) []*junitapi.JUnitTestCase {
+	test := &junitapi.JUnitTestCase{Name: testName}
 	messageRegExp := regexp.MustCompile(regExStr)
 	var failureOutput string
-	var count int
 	for _, event := range events {
 		if isOperatorMatchRegexMessage(event, operatorName, messageRegExp) {
 			// Place the failure time in the message to avoid having to extract the time from the events json file
 			// (in `artifacts) when viewing the test failure output.
 			failureOutput += fmt.Sprintf("%s %s\n", event.From.Format("15:04:05"), event.Message)
-			count++
-		}
-	}
-	test := &junitapi.JUnitTestCase{Name: testName}
-	if count > eventFlakeThreshold {
-		// Flake for now.
-		test.FailureOutput = &junitapi.FailureOutput{
-			Output: failureOutput,
+
+			_, times := getTimesAnEventHappened(fmt.Sprintf("%s - %s", event.Locator, event.Message))
```

Comment:
Was the original issue that we weren't extracting the times & the threshold was too low or just that we weren't extracting the times?  I can restore the previous thresholds if we need.  Also presume that each event / times count is compared in isolation and not accumulated then compared.  Similar to what is in duplicated_events

Reply:
Original issue was that this was happening all the time and we wanted to break out a specific test for the problem that we could flake, but track separately. It was working fine just failing too much.

Reply:
yes, we were failing too much because I didn't extract the count as you're doing above.  Here's the original logic: https://github.com/openshift/origin//blob/b25083353f1495ccb98b11aa78af8442dc559c66/pkg/synthetictests/duplicated_events.go#L341-L343. You can see that the threshold is 20 and not 5.

Reply:
cool, so I will keep the duplicateEventThreshold update as well as this code to extract the times/count to compare against that threshold.

Reply:
Did the original test do anything with "number of times this line occurred"? Or was it just "how many repetitions did we see in each line". 

I'd suggest we stick precisely to whatever logic that had before. It's possible we're only interested in finding events that repeated so close to each other they got compressed by kube together, over 20x.

Reply:
For the original logic it looks like the last group of events over duplicateEventThreshold wins: https://github.com/openshift/origin/blob/0bd61089e01d2c5fcbe03546c69be842fcfd6589/pkg/synthetictests/duplicated_events.go#L336-L365

eventDisplayMessage appears to be generic so the assignment

displayToCount[eventDisplayMessage] = times

ends up being last one wins. 

I thought it might be bugged but with the check for duplicateEventThreshold  if we have a case where it repeated 25 times and then later 5 times we still report the 25 case.  If we have 25 then 21 we report 21 but it will still trigger the flake so good enough.


The way it is rewritten I believe the first occurrence over duplicateEventThreshold will trigger the flake so in the case of multiple groupings {25,21} it could be slightly different message {25) but same result of a flake.